src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php

```diff
@@ -26,163 +26,1921 @@
+            $this->addCSS($ui_path['css'], 'all');
+            $this->addJS($ui_path['js'], false);
+        }
+    }
+
+    /**
+     * Sets the smarty variables and js defs used to show / hide some notifications.
+     */
+    public function initToken(Request $request)
+    {
+        $this->token = $request->query->get('_token');
+    }
+
+    /**
+     * Set breadcrumbs array for the controller page.
+     *
+     * @param array $tabs
+     */
+    public function initBreadcrumbs(array $tabs = [])
+    {
+        if (!is_array($tabs)) {
+            $tabs = [];
+        }
+        $tabId = $this->id;
+        $tabs = Tab::recursiveTab($tabId, $tabs);
+
+        $dummy = ['name' => '', 'href' => '', 'icon' => ''];
+        $breadcrumbs2 = [
+            'container' => $dummy,
+            'tab' => $dummy,
+            'action' => $dummy,
+        ];
+
+        if (!empty($tabs[0])) {
+            $this->addMetaTitle($tabs[0]['name']);
+            $breadcrumbs2['tab']['name'] = $tabs[0]['name'];
+            $breadcrumbs2['tab']['href'] = $this->getContext()->link->getTabLink($tabs[0]);
+            if (!isset($tabs[1])) {
+                $breadcrumbs2['tab']['icon'] = 'icon-' . $tabs[0]['class_name'];
+            }
+        }
+        if (!empty($tabs[1])) {
+            $breadcrumbs2['container']['name'] = $tabs[1]['name'];
+            $breadcrumbs2['container']['href'] = $this->getContext()->link->getTabLink($tabs[1]);
+            $breadcrumbs2['container']['icon'] = 'icon-' . $tabs[1]['class_name'];
+        }
+
+        /* content, edit, list, add, details, options, view */
+        //switch ($this->display) {
+        //    case 'add':
+        //        $breadcrumbs2['action']['name'] = $this->trans('Add', 'Admin.Actions');
+        //        $breadcrumbs2['action']['icon'] = 'icon-plus';
+        //
+        //        break;
+        //    case 'edit':
+        //        $breadcrumbs2['action']['name'] = $this->trans('Edit', 'Admin.Actions');
+        //        $breadcrumbs2['action']['icon'] = 'icon-pencil';
+        //
+        //        break;
+        //    case '':
+        //    case 'list':
+        //        $breadcrumbs2['action']['name'] = $this->trans('List', 'Admin.Actions');
+        //        $breadcrumbs2['action']['icon'] = 'icon-th-list';
+        //
+        //        break;
+        //    case 'details':
+        //    case 'view':
+        //        $breadcrumbs2['action']['name'] = $this->trans('View details', 'Admin.Actions');
+        //        $breadcrumbs2['action']['icon'] = 'icon-zoom-in';
+        //
+        //        break;
+        //    case 'options':
+        //        $breadcrumbs2['action']['name'] = $this->trans('Options', 'Admin.Actions');
+        //        $breadcrumbs2['action']['icon'] = 'icon-cogs';
+        //
+        //        break;
+        //    case 'generator':
+        //        $breadcrumbs2['action']['name'] = $this->trans('Generator', 'Admin.Actions');
```

Comment:
Could you tell me where it appears? It seems new and doesn't look like it's an action.